snappy_pipeline/workflows/somatic_cnv_calling/model.py

```diff
@@ -0,0 +1,232 @@
+
+    minCounts: int = 10
+    """Minimum depth required in the normal for a SNP to be considered"""
+
+    min_BQ: int = 20
+    """Minimum base quality required for a read to be counted"""
+
+    min_MQ: int = 35
+    """Minimum mapping quality required for a read to be counted"""
+
+    skip_any_set: Annotated[
+        str | int | BcftoolsBamFlag | list[str] | list[int] | list[BcftoolsBamFlag],
+        AfterValidator(transform_to_flag),
+    ] = (
+        BcftoolsBamFlag.UNMAP
+        | BcftoolsBamFlag.MUNMAP
+        | BcftoolsBamFlag.SECONDARY
+        | BcftoolsBamFlag.QCFAIL
+        | BcftoolsBamFlag.DUP
+        | BcftoolsBamFlag.SUPPLEMENTARY
+    )
+    """
+    Exclude reads with any on these flags:
+
+    - 2048 0x800 SUPPLEMENTARY supplementary alignment
+    - 1024 0x400 DUP           PCR or optical duplicate
+    -  512 0x200 QCFAIL        not passing filters, such as platform/vendor quality controls
+    -  256 0x100 SECONDARY     secondary alignment
+    -    8 0x008 MUNMAP        next segment in the template unmapped
+    -    4 0x004 UNMAP         segment unmapped
+    """
+    skip_any_unset: Annotated[
+        str | int | BcftoolsBamFlag | list[str] | list[int] | list[BcftoolsBamFlag],
+        AfterValidator(transform_to_flag),
+    ] = BcftoolsBamFlag.PAIRED | BcftoolsBamFlag.PROPER_PAIR
+    """
+    Exclude reads without any on these flags:
+
+    -    2 0x002 PROPER_PAIR each segment properly aligned according to the aligner
+    -    1 0x001 PAIRED      template having multiple segments in sequencing
+    """
+
+    max_depth: int = 8000
+    """Max depth to used in ``bcftools`` (not part of the original implementation)"""
+
+
+class GenomeVersionAscat(enum.StrEnum):
+    GRCh37 = "hg19"
+    GRCh38 = "hg38"
+
+
+class AscatPlatform(enum.StrEnum):
+    AffySNP6 = "AffySNP6"
+    Custom10k = "Custom10k"
+    IlluminaASA = "IlluminaASA"
+    IlluminaGSAv3 = "IlluminaGSAv3"
+    Illumina109k = "Illumina109k"
+    IlluminaCytoSNP = "IlluminaCytoSNP"
+    IlluminaCytoSNP850k = "IlluminaCytoSNP850k"
+    Illumina610k = "Illumina610k"
+    Illumina660k = "Illumina660k"
+    Illumina700k = "Illumina700k"
+    Illumina1M = "Illumina1M"
+    Illumina25M = "Illumina2.5M"
+    IlluminaOmni15 = "IlluminaOmni15"
+    IlluminaGDACyto8 = "IlluminaGDACyto-8"
+    Affy10k = "Affy10k"
+    Affy100k = "Affy100k"
+    Affy250k_sty = "Affy250k_sty"
+    Affy250k_nsp = "Affy250k_nsp"
+    AffyOncoScan = "AffyOncoScan"
+    AffyCytoScanHD = "AffyCytoScanHD"
+    HumanCNV370quad = "HumanCNV370quad"
+    HumanCore12 = "HumanCore12"
+    HumanCoreExome24 = "HumanCoreExome24"
+    HumanOmniExpress12 = "HumanOmniExpress12"
+    IlluminaOmniExpressExome = "IlluminaOmniExpressExome"
+    WGS_hg_50X = "WGS_hg_50X"
+
+
+class AscatAdvancedParams(SnappyModel):
+    penalty: float = 70.0
+    """Segmentation penalty"""
+
+    gamma: float = 1.0
+    """Compactation of LogR profiles. Must be set to 1 for WGS, WES & Panel"""
```

Comment:
Is this validated?

Reply:
No, it's not, it is a recommendation from the `ASCAT` authors. The parameter is mostly used with array data, as far as I understand.
Should I remove this parameter? (Personally, I like to keep all of them, not start choosing, because one day we might find a case where we need to modify it)